presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2215,6 +2215,14 @@ public void testJoinWithComplexExpressions3()
     }
 
     @Test
+    public void testJoinWithNormalization() {
+        assertQuery("select COUNT(*) from nation a join nation b on not ((a.nationkey + b.nationkey) <> b.nationkey)");
+        assertQuery("select COUNT(*) from nation a join nation b on not (a.nationkey <> b.nationkey)");
+        assertQuery("select COUNT(*) from nation a join nation b on not (a.nationkey = b.nationkey)");
```

Comment:
In addition to that test, add one with nested `not` `(not(not x = y))` because normalization takes care of it as well.